src/SnowflakeTransformation.php

```diff
@@ -0,0 +1,148 @@
+
+        if ($config->getQueryTimeout()) {
+            $sessionVariables['STATEMENT_TIMEOUT_IN_SECONDS'] = $config->getQueryTimeout();
+        }
+
+        array_walk($sessionVariables, function (&$item, $key): void {
+            $item = vsprintf(
+                '%s=%s',
+                [
+                    $key,
+                    $item,
+                ]
+            );
+        });
+
+        $query = sprintf('ALTER SESSION SET %s;', implode(',', $sessionVariables));
+        $this->runRetriableQuery($query, 'alter session');
+    }
+
+    public function processSteps(array $steps): void
+    {
+        foreach ($steps as $step) {
+            if ($step['execution'] === 'parallel') {
+                foreach ($step['blocks'] as $block) {
+                    $this->logger->info(sprintf('Start process block "%s"', $block['name']));
+                    $this->executionQueries($block['name'], $block['script']);
+                }
+            } else {
+                foreach ($step['blocks'] as $block) {
+                    $this->logger->info(sprintf('Start process block "%s"', $block['name']));
+                    $this->executionQueries($block['name'], $block['script']);
+                }
+            }
+        }
+    }
+
+    public function executionQueries(string $name, array $queries): void
+    {
+        foreach ($queries as $query) {
+            $this->logger->info(sprintf('Run query: %s', $query));
+            $this->runRetriableQuery($query, $name);
+        }
+    }
+
+    private function createConnection(): Connection
+    {
+        $connection = new Connection($this->databaseConfig);
+        try {
+            if (!empty($this->databaseConfig['schema'])) {
+                $connection->query(
+                    sprintf(
+                        'USE SCHEMA %s',
+                        QueryBuilder::quoteIdentifier($this->databaseConfig['schema'])
+                    )
+                );
+            }
+        } catch (CannotAccessObjectException $e) {
+            throw new UserException($e->getMessage(), 0, $e);
+        }
+
+        return $connection;
+    }
+
+    private function runRetriableQuery(string $query, string $errorMessage): void
+    {
+        $retryPolicy = new SimpleRetryPolicy(
+            SimpleRetryPolicy::DEFAULT_MAX_ATTEMPTS,
+            ['PDOException', 'ErrorException']
+        );
+        $backoffPolicy = new ExponentialBackOffPolicy();
+        $retryProxy = new RetryProxy($retryPolicy, $backoffPolicy);
+
+        try {
+            $retryProxy->call(function () use ($query): void {
+                try {
+                    $this->connection->query($query);
+                } catch (\Throwable $exception) {
+                    $this->tryReconnect();
+                    throw $exception;
+                }
+            });
+        } catch (\Throwable $exception) {
+            $message = sprintf('[%s] Query failed: "%s"', $errorMessage, $query);
```

Comment:
```suggestion
            $message = sprintf('Query "%s" failed with error "%s".', $errorMessage, $query);
```

kazdej pes jina ves!

Reply:
upraveno